smslibrary/src/main/java/com/eis/smslibrary/SMSMessageHandler.java

```diff
@@ -49,10 +50,11 @@ public void setMessageParseStrategy(@NonNull final MessageParseStrategy<String,S
      * @return the message if the string has been parsed correctly, null otherwise
      */
     public SMSMessage parseMessage(@NonNull final String peerData, @NonNull final String messageData){
-        if(SMSPeer.checkPhoneNumber(peerData) != SMSPeer.TelephoneNumberState.TELEPHONE_NUMBER_VALID)
-            return null;
+        SMSPeer peer;
+        try { peer = new SMSPeer(peerData); }
+        catch(InvalidTelephoneNumberException | IllegalArgumentException e) { return null; }
 
-        return parseStrategy.parseMessage(messageData,new SMSPeer(peerData));
+        return parseStrategy.parseMessage(messageData,peer);
```

Comment:
Good way to do it, but I would have done it like this
```
try {
        SMSPeer peer = new SMSPeer(peerData);
        return parseStrategy.parseMessage(messageData,peer);
}
catch(InvalidTelephoneNumberException | IllegalArgumentException e) { return null; }
```
Just to separate into 2 distinct block: everything goes well and something doesn't work

Reply:
Right, more logical, I'll fix it